### ISSUE TITLE
Error: Plugin did not respond

### ISSUE THREAD

**Issue by julian-vp** [dynatrace-oss/terraform-provider-dynatrace#244]
**Describe the bug**
Using terraform-provider-dynatrace_v1.28.0, when referencing `dynatrace_alerting_profiles` data, we started to get the below error in the pipeline. 

It used to work fine last week.

│ Error: Plugin did not respond
│ 
│   with data.dynatrace_alerting_profiles.this,
│   on vars.tf line 262, in data "dynatrace_alerting_profiles" "this":
│  262: data "dynatrace_alerting_profiles" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.

`Stack trace from the terraform-provider-dynatrace_v1.28.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc607c1]

goroutine 161 [running]:
github.com/dynatrace-oss/terraform-provider-dynatrace/datasources/alerting.DataSourceRead(0x0?, {0xf1ed20?, 0xc00028b7c0})
	github.com/dynatrace-oss/terraform-provider-dynatrace/datasources/alerting/data_source.go:116 +0x581
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x14009a8?, {0x14009a8?, 0xc000b11c50?}, 0xd?, {0xf1ed20?, 0xc00028b7c0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0001e29a0, {0x14009a8, 0xc000b11c50}, 0xc0009ec300, {0xf1ed20, 0xc00028b7c0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000010240, {0x14009a8?, 0xc000b11b00?}, 0xc000d0c040)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1195 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc00028bc20, {0x14009a8?, 0xc000b11320?}, 0xc000712dc0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:658 +0x403
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x11b1e00?, 0xc00028bc20}, {0x14009a8, 0xc000b11320}, 0xc0002630a0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002d81e0, {0x141dc80, 0xc000204000}, 0xc0006d50e0, 0xc0005f80f0, 0x1bdd6b0, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0002d81e0, {0x141dc80, 0xc000204000}, 0xc0006d50e0, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-dynatrace_v1.28.0 plugin crashed!

`

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @julian-vp ,

Thanks for reaching out to us. We are planning to release v1.29.1 early next week that won't crash anymore at that point.
Would you be willing to run your module again using the environment variables

```
DYNATRACE_LOG_HTTP=terraform-provider-dynatrace.http.log
DYNATRACE_HTTP_RESPONSE=true
```

It would give us insight into what's special in your environment regarding management zones. The line number of where the crash happens tells me, that there exists a management zone with an `ID` where the (so-called) `legacyID` cannot get extracted from. The Plugin would produce a file `terraform-provider-dynatrace.http.log` with all the HTTP communication logged. We don't need to use a public GitHub issue for exchanging that log file - we can use email, of course.

**Comment by julian-vp** [comment]
Hello @Dynatrace-Reinhard-Pilz 
Thank you for your prompt response. We will try again with release v1.29.1 once it is available.

**Comment by julian-vp** [comment]
Hi @Dynatrace-Reinhard-Pilz 
Do you have any updates regarding v1.29.1?

I also tried to run the plan it with the provided environment variables but I am not getting the log file generated.

**Comment by julian-vp** [comment]
Hi @Dynatrace-Reinhard-Pilz 
I can confirm that the issues has been fixed in v1.30.0

Thanks
